### PR TITLE
fix warning in endian.h

### PR DESCRIPTION
**Pull request by dimas1185**

### CODE REVIEW

**Review by larryk85 — APPROVED** [review]

File: include/endian.h [hunk 1]
```diff
@@ -29,12 +29,12 @@ static __inline uint16_t __bswap16(uint16_t __x)
...
 static __inline uint32_t __bswap32(uint32_t __x)
 {
-	return __x>>24 | __x>>8&0xff00 | __x<<8&0xff0000 | __x<<24;
+	return __x>>24 | (__x>>8)&0xff00 | (__x<<8)&0xff0000 | __x<<24;
 }
 
 static __inline uint64_t __bswap64(uint64_t __x)
 {
-	return __bswap32(__x)+0ULL<<32 | __bswap32(__x>>32);
+	return (__bswap32(__x)+0ULL)<<32 | __bswap32(__x>>32);
```
Comment (greg7mdp) [inline include/endian.h]: I thing I'd prefer this:
```suggestion
	return uint64_t(__bswap32(__x))<<32 | __bswap32(__x>>32);
```
In c++23, we'll have [this](https://en.cppreference.com/w/cpp/numeric/byteswap) finally!

Comment (dimas1185) [reply include/endian.h]: `uint64_t` is not defined in that scope, I'll try to make it simple

Comment (greg7mdp) [reply include/endian.h]: Are you sure?
image

Comment (dimas1185) [reply include/endian.h]: ooops, it was misleading error, thank you

**Pull request merged**